src/Carbon/Carbon.php

```diff
@@ -2847,10 +2858,13 @@ public function diffForHumans(Carbon $other = null, $absolute = false, $short =
 
         $transId = $isNow ? ($isFuture ? 'from_now' : 'ago') : ($isFuture ? 'after' : 'before');
 
-        // Some langs have special pluralization for past and future tense.
-        $tryKeyExists = $unit.'_'.$transId;
-        if ($tryKeyExists !== static::translator()->transChoice($tryKeyExists, $count)) {
-            $time = static::translator()->transChoice($tryKeyExists, $count, array(':count' => $count));
+        if ($parts === 1) {
+            // Some langs have special pluralization for past and future tense.
+            $key = $unit.'_'.$transId;
```

Comment:
`$unit` might not be set at this point, so it probably needs to be initiated somewhere earlier in the function?